core.lua

```diff
@@ -1066,6 +1066,113 @@ do
+		local overallBest = { dungeon = profile.maxDungeon, level = profile.dungeons[profile.maxDungeon.index] }	-- overall best
+
+		if addLFD then
+			local hasArgs, dungeonIndex = ...
+			if hasArgs == true then
+				best.dungeon = CONST_DUNGEONS[dungeonIndex] or best.dungeon
+			end
+		end
+
+		if focusDungeon then
+			local hasArgs, dungeonIndex = ...
+			if hasArgs == true then
+				best.dungeon = CONST_DUNGEONS[dungeonIndex] or best.dungeon
+			end
+		end
+
+		if focusKeystone then
+			local hasArgs, arg1, arg2 = ...
+			if hasArgs == true then
+				if ns.DEBUG_MODE then -- TODO
+					table.insert(lines, "focusKeystone arg1 = " .. tostring(arg1))
+					table.insert(lines, "focusKeystone arg2 = " .. tostring(arg2))
+				end
+			end
+		end
+
+		-- if we don't have a best dungeon focused by this point, try to find one based on our queue or current instance
+		if not best.dungeon and addLFD then
+			local numSigned, status = GetLFDStatus()
+			if numSigned then
+				if numSigned == true then
+					best.dungeon = status.dungeon
+					best.level = status.level or 0
+				elseif numSigned > 0 then
+					local highestDungeon
+					for j = 1, numSigned do
+						local d = status[j]
+						if not highestDungeon or d.level > highestDungeon.level then
+							highestDungeon = d
+						end
+					end
+					best.dungeon = highestDungeon
+					best.level = highestDungeon.level or 0
+				end
+			end
+			if not best.dungeon then
+				best.dungeon = GetInstanceStatus()
+			end
+		end
+
+		-- if we have a dungeon, but no level assigned to it, try to read one from our profile
+		if best.dungeon and (not best.level or best.level < 1) then
+			best.level = profile.dungeons[best.dungeon.index] or 0
+		end
+
+		-- if no dungeon, or the level is undefined or 0, drop showing both as it's irrelevant information
+		if not best.dungeon or (best.level and best.level < 1) then
+			best.dungeon, best.level = nil, 0
+		end
+
+		if profile.isFifteenPlusAchievement then
+			if profile.maxDungeonLevel < 15 then
+				best.text = L.KEYSTONE_COMPLETED_15
+			end
+		elseif profile.isTenPlusAchievement then
+			if profile.maxDungeonLevel < 10 then
+				best.text = L.KEYSTONE_COMPLETED_10
+			end
+		elseif profile.isFivePlusAchievement then
+			if profile.maxDungeonLevel < 5 then
+				best.text = L.KEYSTONE_COMPLETED_5
+			end
+		end
+
+		if not best.text and (not best.dungeon or overallBest.dungeon.index ~= best.dungeon.index) and overallBest.level > 0 then
+			local bestRunLabel = (isProfile or not ns.addonConfig.showBestRunFirst) and L.BEST_RUN or L.RAIDERIO_BEST_RUN
+			local lineColor = (isProfile or not ns.addonConfig.showBestRunFirst) and {1, 1, 1} or {1, 0.85, 0}
+			table.insert(lines, {bestRunLabel, GetStarsForUpgrades(profile.dungeonUpgrades[overallBest.dungeon.index]) .. overallBest.level .. " " .. overallBest.dungeon.shortNameLocale, lineColor[1], lineColor[2], lineColor[3], GetScoreColor(profile.allScore)})
+		end
+
+		if best.dungeon then
```

Comment:
Maybe wait a bit for the merge, I would like to check these conditions a bit more (and maybe refactor)